tests/index.test.ts

```diff
@@ -3,61 +3,166 @@ import * as test from 'tape';
+        ])
+    });
+
+    t.deepEqual(
+        reporter(
+            Unions.decode({
+                oneOf: '',
+                stringUnion: '',
+                interfaceUnion: ''
+            })
+        ),
+        [
+            [
+                'Expecting one of:',
+                '    { key: string }',
+                '    { code: number }',
+                'at interfaceUnion but instead got: ""'
+            ].join('\n'),
+            'Expecting "a" | "b" | "c" at oneOf but instead got: ""',
+            [
+                'Expecting one of:',
+                '    "a"',
+                '    "b"',
+                '    "c"',
+                'at stringUnion but instead got: ""'
+            ].join('\n')
+        ]
+    );
+
+    t.deepEqual(
+        reporter(
+            Unions.decode({
+                oneOf: 'a',
+                stringUnion: 'a',
+                interfaceUnion: {}
+            })
+        ),
+        [
+            [
+                'Expecting one of:',
+                '    { key: string }',
+                '    { code: number }',
+                'at interfaceUnion but instead got: {}'
+            ].join('\n')
+        ]
+    );
+
+    t.deepEqual(
+        reporter(
+            Unions.decode({
+                oneOf: 'a',
+                stringUnion: 'a',
+                interfaceUnion: { code: '123' }
+            })
+        ),
+        [
+            [
+                'Expecting one of:',
+                '    { key: string }',
+                '    { code: number }',
+                'at interfaceUnion but instead got: {"code":"123"}'
+            ].join('\n')
+        ]
+    );
+
+    const Gender = iots.union([
+        iots.literal('Male'),
+        iots.literal('Female'),
+        iots.literal('Other')
+    ]);
     const Person = iots.interface({
         name: iots.string,
-        age: iots.number,
         gender: Gender,
-        children: iots.array(
-            iots.interface({
-                gender: Gender,
-            })
-        )
-    });
-    const result = Person.decode({
-        name: 'Giulio',
-        children: [{ gender: 'Whatever' }]
+        children: iots.array(iots.interface({ gender: Gender }))
     });
 
-    t.deepEqual(reporter(result), [
-        'Expecting number at age but instead got: undefined.',
-        'Expecting "Male" at gender.0 but instead got: undefined.',
-        'Expecting "Female" at gender.1 but instead got: undefined.',
-        'Expecting "Male" at children.0.gender.0 but instead got: "Whatever".',
-        'Expecting "Female" at children.0.gender.1 but instead got: "Whatever".'
+    t.deepEqual(reporter(Person.decode({ name: 'Jane', children: [] })), [
+        'Expecting number at age but instead got: undefined',
+        [
+            'Expecting one of:',
+            '    "Male"',
+            '    "Female"',
+            '    "Other"',
+            'at gender but instead got: undefined'
+        ].join('\n')
```

Comment:
@osdiab this case should cover the exampled you mentioned on https://github.com/OliverJAsh/io-ts-reporters/pull/17#issuecomment-631471440

Expected:
```
Expecting one of:
    "Male"
    "Female"
    "Other"
at gender but instead got: undefined
```

But currently it outputs:
```
Expecting one of:
    "Male"
    "Female"
    "Other"
    "Male"
    "Female"
    "Other"
    "Male"
    "Female"
    "Other"
at gender but instead got: undefined
```